website/static/js/contribRemover.js

```diff
@@ -0,0 +1,275 @@
+var Raven = require('raven-js');
+
+var oop = require('./oop');
+var $osf = require('./osfHelpers');
+var Paginator = require('./paginator');
+
+var RemoveContributorViewModel = oop.extend(Paginator, {
+    constructor: function(title, nodeId, userName, userId, contribShouter, pageChangedShouter) {
+        this.super.constructor.call(this);
+        var self = this;
+        self.title = title;
+        self.nodeId = nodeId;
+        self.userId = userId;
+        self.contributorToRemove = ko.observable('');
+        self.REMOVE = 'remove';
+        self.REMOVE_ALL = 'removeAll';
+        self.REMOVE_NO_CHILDREN = 'removeNoChildren';
+        self.REMOVE_SELF = 'removeSelf';
+
+        //This shouter allows the ContributorsViewModel to share which contributor to remove
+        // with the RemoveContributorViewModel
+        contribShouter.subscribe(function(newValue) {
+            self.contributorToRemove(newValue);
+        }, self, 'contribMessageToPublish');
+
+        //This shouter allows RemoveContributorViewModel to know if the
+        // ContributorsViewModel is in a dirty state to prevent removal
+        self.pageChanged = ko.observable(false);
+        pageChangedShouter.subscribe(function(newValue) {
+            self.pageChanged(newValue);
+        }, self, 'changedMessageToPublish');
+
+        self.page = ko.observable(self.REMOVE);
+        self.pageTitle = ko.computed(function() {
+            return {
+                remove: 'Delete Contributor',
+                removeAll: 'Delete Contributor',
+                removeNoChildren: 'Delete Contributor'
+            }[self.page()];
+        });
+        self.userName = ko.observable(userName);
+        self.deleteAll = ko.observable(false);
+        var nodesOriginal = {};
+        self.nodesOriginal = ko.observableArray();
+        self.loadingSubmit = ko.observable(false);
+
+        /*
+         *   To remove, a contributor, you need one bibliographic contributor
+         *   (visibleContributors) one admin, and one registered contributor.
+         *   So we make new arrays with the contributor removed, and see if
+         *   the results are > 0.
+         */
+        self.canRemoveNodes = ko.computed(function() {
+            var canRemoveNodes = {};
+            var nodesOriginalLocal = ko.toJS(self.nodesOriginal());
+            if (self.contributorToRemove()) {
+                var contributor_id = self.contributorToRemove().id;
+                for (var key in nodesOriginalLocal) {
+                    var node = nodesOriginalLocal[key];
+                    var contributorOnNode = false;
+                    //User cannot modify the node without admin permissions.
+                    if (node.isAdmin || self.removeSelf) {
+                        for (var i = 0; i < node.contributors.length; i++) {
+                            if (node.contributors[i].id === self.contributorToRemove().id) {
+                                contributorOnNode = true;
```

Comment:
Once you've made it here, there no reason to continue iterating, so you can break out here.